File: Chap6/C/exo3_b.c
#include<stdio.h>
#include<fcntl.h>
#include<stdlib.h>
#include<string.h>
#include<unistd.h>

int *chargement_en_memoire(char *name,int *taille)
{
    int fd=open(name,O_RDONLY);
    int *tab=NULL;
    int *temp;
    int nombre,i=0;
    while(read(fd,&nombre,sizeof(int))!=0)//lire tant que le fichier est vide mais par int 
    {
        temp = realloc(tab, (i + 1) * sizeof(int));
        if (temp == NULL)
        {
            free(tab);
            return NULL;
        }

        tab = temp;
        tab[i] = nombre;
        i++;
    }
    *taille=i-1;
    return(tab);
}
int main()
{
    char fichier[100];
    int taille=0;
    printf("Entrer le nom du fichier ");
    scanf("%s",fichier);
    int *tab=chargement_en_memoire(fichier,&taille);
    int i;
    for(i=0;i<taille;i++)
    {
        printf("%d\n",tab[i]);
    
    }
}

File: Chap6/C/exo3_b.py
import os
import struct

def chargement_en_memoire(fichier,reference):
    tab=[]
    fd=os.open(fichier,os.O_RDONLY)
    data=os.read(fd, 4)
    while data:
      nombre=struct.unpack("i", data)[0] #convertit bytes en int   
      tab.append(nombre)
      data=os.read(fd, 4)
    os.close(fd)        
    return tab


fichier=input("Entrer le nom du fichier binaire à charger en mémoire :")
taille=input("Entrer le nombre d'élement que vous vouliez voir :")
tab=chargement_en_memoire(fichier,reference)
for i in range(taille):
   print(tab[i],"\n")